### PR TITLE
Simulator categories

### LINKED ISSUE #9: As a user, I want to be able to see what entries in the results are simulators and what are plugins/frontends/frameworks
So that as a user, I can understand what tool is doing the simulation, and what tool is sort of an "add-on" that enables the use of the tool.

This will bring up relations between tools, and we also need to think of a way to filter/distinguish these in the frontend

### PR DESCRIPTION
**Pull request by mstimberg**
Closes #9

Not quite in time for the milestone, but this PR adds little "category icons" to the simulator buttons and allows filtering by them. It turned out to be more complicated than what I hoped, since the standard panel `Button` widget does only allow a single icon and we can only pass in only raw text as the label (ie. no `<i class="..."></i>` to include an icon or similar). I hacked something together with a `ReactiveHTML` component, which works fine in general, but for some reason it doesn't have access to the usual stylesheet that styles the button. I now add this manually in the `categories.css`. There's certainly some cleanup needed for the code, and I quickly filled out categories for all tools which means some of them are certainly wrong and others up for debate, but I guess it is good enough™?

Note that you need to run `panel` with `--static-dirs assets=./assets` now, so that the web server has access to the CSS file – I hope I updated this info everywhere.

With this, I think it looks good enough for the 0.1 and we can finish the milestone :champagne: :tada: ? See example screenshot below:

Screenshot 2023-06-29 at 16-21-04 SimSelect

### CODE REVIEW

File: simtools/Arbor-Playground.yaml [hunk 1]
```diff
@@ -1,4 +1,5 @@
 - name: Arbor Playground
+- categories: frontend, backend
```
Comment (brenthuisman) [inline simtools/Arbor-Playground.yaml:2]:
```suggestion
- categories: frontend
```

File: simtools/Brain-Scaffold-Builder.yaml [hunk 2]
```diff
@@ -1,4 +1,5 @@
 - name: Brain Scaffold Builder
+- categories: frontend, backend
```
Comment (brenthuisman) [inline simtools/Brain-Scaffold-Builder.yaml:2]:
```suggestion
- categories: frontend
```

**Review by brenthuisman — COMMENTED** [review]
ATM there's just two categories, and I'm not so sure they are clear: a tool like BSB is a frontend to simulators, but it's also possible to integrate into something else, i.e. maybe it's a backend too? The Issue says these two categories are about distinguishing simulation engines from wrappers, in which case BSB is a wrapper, and a tool can not have both labels, because it's silly to wrap yourself :P

**Comment by mstimberg** [comment]
> ATM there's just two categories, and I'm not so sure they are clear: a tool like BSB is a frontend to simulators, but it's also possible to integrate into something else, i.e. maybe it's a backend too? The Issue says these two categories are about distinguishing simulation engines from wrappers, in which case BSB is a wrapper, and a tool can not have both labels, because it's silly to wrap yourself :P

We should certainly discuss all this in more detail – I'd say the tools like BSB are probably the most tricky (maybe `frontend, tool` would be the most appropriate?) For me, everything that does not actually run the simulation itself (could be a GUI, but also something like NetPyNE)  is a `frontend`, everything that can execute a simulation but is meant to use with something else for defining what gets simulated is a `backend` (coreNeuron, Brian2CUDA, etc.). Simulators like Arbor, Nest, Neuron, Brian can do both, so they'd be `frontend, backend`. 

Regarding the suggestions for better frontend/backend labels, I agree (but wait for feedback from others before changing them). For `backend`, "compute engine" might work, too.

**Comment by sanjayankur31** [comment]
How about just "simulator" for tools that do actually simulate? "Compute engine" etc. could mean different things to different people---technically, one can use python to compute, so it's a "compute engine". "frontend" "backend" is too jargony, not everyone would know what they mean?

So yeh, tools that simulate get a "simulator" tag, APIs can get either "interface" or "user library", and others get other tags: "standard", "utility", "misc", "other"?

**Comment by mstimberg** [comment]
I get the jargon criticism, but using "simulator" for the simulation engines could be misleading as well, no? If we call something like coreNeuron or Brian2CUDA a "simulator", would users understand that they cannot use them on their own? On the other hand, maybe we shouldn't do the "frontend + backend" thing for full simulator? So instead of (ignoring the specific terms for now) "frontend", "backend", "frontend, backend", we'd have "frontend", "backend", "simulator"? But then, from a filtering perspective, a search for "frontend" should also show the "simulator"…? Maybe confusing as well.

**Comment by sanjayankur31** [comment]
Hrm, would Brian2CUDA be considered a simulator though? It doesn't simulate :thinking: 

What if this is the wrong axis to think of it since tools can belong to different "categories" here? Instead of saying what category a tool includes, would it make sense to list what features a tool provides?

Brian:
- features:
  - simulation engine
  - python API


The idea is to let people search/filter. So if they filter by "simulation engine", they get all tools that provide this feature but may also provide other features like APIs, GUIs and so on? Between features and relations, queries should cover tools that a user should be looking into?

**Comment by mstimberg** [comment]
> Hrm, would Brian2CUDA be considered a simulator though? It doesn't simulate

Ok, now I am confused what you mean by "simulate" :sweat_smile: It's responsible for running the simulation on the GPU, so it is the "compute engine" used by Brian (for running things on the GPU).



> The idea is to let people search/filter. So if they filter by "simulation engine", they get all tools that provide this feature but may also provide other features like APIs, GUIs and so on? Between features and relations, queries should cover tools that a user should be looking into?

Actually, that's the way it works currently, and I'd be happy to call them features instead (this word is only used in the code and yaml file at the moment).

**Comment by sanjayankur31** [comment]
Hrm, yeh, in that case it is a "compute engine". "features" feels more flexible to me. As a user, it's more naturally to ask "what can tool do" vs "what is it", if that makes sense?

**Comment by mstimberg** [comment]
Internally, it now uses `features` instead of `categories` everywhere, and I used "user interface" instead of "interface/frontend", and "compute engine" instead of "backend/engine". In the yaml file, these features are still called "frontend", "backend", but that should be ok for now, I guess.

**Review by ree-gupta — APPROVED** [review]
I do not have many opinions on the semantics of the “categories” but on the UI end, I think it looks pretty nice and makes it more user-friendly!

**Pull request merged**